client/signup/main.jsx

```diff
@@ -45,7 +45,8 @@ import * as oauthToken from 'lib/oauth-token';
 /**
  * Constants
  */
-const MINIMUM_TIME_LOADING_SCREEN_IS_DISPLAYED = 8000;
+const MINIMUM_TIME_LOADING_SCREEN_IS_DISPLAYED_FREE = 8000;
+const MINIMUM_TIME_LOADING_SCREEN_IS_DISPLAYED_PREMIUM = 3000;
```

Comment:
After reading p3Ex-1tC-p2, I think three seconds are enough as we don't want the users to click on the "confirm email" link before going to checkout.